rust/processor/src/processors/marketplaces_processor.rs

```diff
@@ -0,0 +1,1022 @@
+                    );
+                }
+
+                if let Some(activity) = activity {
+                    marketplace_activities.push(activity);
+                }
+                if let Some(listing) = MarketplaceListing::from_event(
+                    &event,
+                    txn_version,
+                    txn_timestamp,
+                    &entry_function_id_str,
+                    &marketplace_contract_address,
+                    &coin_type,
+                )
+                .unwrap()
+                {
+                    nft_listings.push(listing);
+                }
+                if let Some(token_offer) = MarketplaceTokenOffer::from_event(
+                    &event,
+                    txn_version,
+                    txn_timestamp,
+                    &entry_function_id_str,
+                    &marketplace_contract_address,
+                    &coin_type,
+                )
+                .unwrap()
+                {
+                    nft_token_offers.push(token_offer);
+                }
+                if let Some(auction) = MarketplaceAuction::from_event(
+                    &event,
+                    txn_version,
+                    txn_timestamp,
+                    &entry_function_id_str,
+                    &marketplace_contract_address,
+                    &coin_type,
+                )
+                .unwrap()
+                {
+                    nft_auctions.push(auction);
+                }
+                if let Some(collection_offer) = MarketplaceCollectionOffer::from_event(
+                    &event,
+                    txn_version,
+                    txn_timestamp,
+                    &entry_function_id_str,
+                    &marketplace_contract_address,
+                    &coin_type,
+                )
+                .unwrap()
+                {
+                    // The collection offer resource may be deleted after it is filled,
+                    // so we need to parse collection offer metadata from the event
+                    let collection_offer_filled_metadata = CollectionOfferEventMetadata {
+                        collection_offer_id: collection_offer.collection_offer_id.clone(),
+                        fee_schedule_id: collection_offer.fee_schedule_id.clone(),
+                        buyer: collection_offer.buyer.clone(),
+                        item_price: collection_offer.item_price.clone(),
+                        collection_metadata: collection_metadata.clone().unwrap(),
+                    };
+                    collection_offer_filled_metadatas.insert(
+                        collection_offer_filled_metadata.collection_offer_id.clone(),
+                        collection_offer_filled_metadata,
+                    );
+
+                    nft_collection_offers.push(collection_offer);
+                }
+            }
+
+            let mut object_metadatas: HashMap<String, ObjectCore> = HashMap::new();
```

Comment:
are you removing code below? this seems like it should be in the model files.

Reply:
Not sure what is the best way to handle the code below, these are HashMap and being used to store data we aggregate from wsc Loop 2 below, then we restruct them in Loop 3. 

One idea I have, is to define a model that has all `Option` field, and we construct the model/object as we go on Loop 2?

cc: Wonder if @rtso has any good idea to simplify this?